nativepython/tests/string_compilation_test.py

```diff
@@ -457,3 +457,67 @@ def test_find_perf(f, s, subs):
                 .format(c_elapsed, elapsed)
         )
         """
+
+    def check_string_join(self, f):
+        """This function contains tests for implementation of the string join
+        for both: ListOf(str) and TupleOf(str).
+        The tests for the types are as separate functions, so it will be easier
+        to find the failing line later.
+        """
+        res = f(",", ListOf(str)(["1", "2", "3"]))
+        self.assertEqual("1,2,3", res)
+
+        # test longer separator
+        res = f("---", ListOf(str)(["1", "2", "3"]))
+        self.assertEqual("1---2---3", res)
+
+        # test longer items
+        res = f("---", ListOf(str)(["aaa", "bb", "c"]))
+        self.assertEqual("aaa---bb---c", res)
+
+        # test empty separator
+        res = f("", ListOf(str)(["1", "2", "3"]))
```

Comment:
need also to test some empty strings in the list itself

Reply:
done